nilearn/plotting/surface/_matplotlib_backend.py

```diff
@@ -0,0 +1,810 @@
+
+    This function computes the tick values for the colorbar.
+    """
+    # Default number of ticks is 5...
+    n_ticks = 5
+    # ...unless we are dealing with integers with a small range
+    # in this case, we reduce the number of ticks
+    if cbar_tick_format == "%i" and vmax - vmin < n_ticks - 1:
+        return np.arange(vmin, vmax + 1)
+    else:
+        return get_cbar_ticks(vmin, vmax, threshold, n_ticks)
+
+
+def _get_view_plot_surf(hemi, view):
+    """Help function for plot_surf with matplotlib engine.
+
+    This function checks the selected hemisphere and view, and
+    returns elev and azim.
+    """
+    check_views([view])
+    check_hemispheres([hemi])
+    if isinstance(view, str):
+        if hemi == "both" and view in ["lateral", "medial"]:
+            raise ValueError(
+                "Invalid view definition: when hemi is 'both', "
+                "view cannot be 'lateral' or 'medial'.\n"
+                "Maybe you meant 'left' or 'right'?"
+            )
+        return MATPLOTLIB_VIEWS[hemi][view]
+    return view
+
+
+def _rescale(data, vmin=None, vmax=None):
+    """Rescales the data."""
+    data_copy = np.copy(data)
+    # if no vmin/vmax are passed figure them out from data
+    vmin, vmax = _get_bounds(data_copy, vmin, vmax)
+    data_copy -= vmin
+    data_copy /= vmax - vmin
+    return data_copy, vmin, vmax
+
+
+def _threshold_and_rescale(data, threshold, vmin, vmax):
+    """Help for plot_surf.
+
+    This function thresholds and rescales the provided data.
+    """
+    data_copy, vmin, vmax = _rescale(data, vmin, vmax)
+    return data_copy, _threshold(data, threshold, vmin, vmax), vmin, vmax
+
+
+def _threshold(data, threshold, vmin, vmax):
+    """Thresholds the data."""
+    # If no thresholding and nans, filter them out
+    if threshold is None:
+        mask = np.logical_not(np.isnan(data))
+    else:
+        mask = np.abs(data) >= threshold
+        if vmin > -threshold:
+            mask = np.logical_and(mask, data >= vmin)
+        if vmax < threshold:
+            mask = np.logical_and(mask, data <= vmax)
+    return mask
+
+
+class MatplotlibBackend(SurfaceBackend):
+    @property
+    def name(self):
+        return "matplotlib"
+
+    def _plot_surf(
+        self,
+        coords,
+        faces,
+        surf_map=None,
+        bg_map=None,
+        hemi="left",
+        view=None,
+        cmap=None,
+        symmetric_cmap=None,
+        colorbar=False,
+        avg_method=None,
+        threshold=None,
+        alpha=None,
+        bg_on_data=False,
+        darkness=0.7,
+        vmin=None,
+        vmax=None,
+        cbar_vmin=None,
+        cbar_vmax=None,
+        cbar_tick_format="auto",
+        title=None,
+        title_font_size=None,
+        output_file=None,
+        axes=None,
+        figure=None,
+    ):
+        parameters_not_implemented = {
+            "symmetric_cmap": symmetric_cmap,
+            "title_font_size": title_font_size,
+        }
```

Comment:
This did not exist in original code, let me know if it is not ok.

Reply:
Gives warning as plotly backend gives warnings for the parameters not implemented for plotly.